legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/PureGrammarComposer.java

```diff
@@ -105,6 +107,55 @@ public String renderPureModelContextData(PureModelContextData pureModelContextDa
         return composedSections.select(section -> !section.isEmpty()).makeString("\n\n");
     }
 
+    public Map<String, String> renderPureModelContextDataForEachElement(PureModelContextData pureModelContextData)
```

Comment:
These method still seem rather clunky to me. Will it be simpler if we do something like:
1. Read the section index if any, transform section index into 1 section per element (with the import of the whole section)
2. Create an index of element path by section (we would need to handle duplicate somehow)
3. Go through that index and compose each element

I felt like the logic is much simpler that way

Also, since we're doing it, there's a case that we haven't thought about where users switch from single text-mode to multiple-file text-mode. When they go back, their sections would be broken down completely and import paths would be duplicated every where. What should we do here? @gayathrir11 and @MauricioUyaguari ?